pkg/storage/unified/search/bleve.go

```diff
@@ -481,12 +481,22 @@ var textSortFields = map[string]string{
 }
 
 // Convert a "requirement" into a bleve query
-func requirementQuery(req *resource.Requirement, prefix string) (query.Query, *resource.ErrorResult) {
+func requirementQuery(req *resource.Requirement, prefix string, opts *resource.ListOptions) (query.Query, *resource.ErrorResult) {
 	switch selection.Operator(req.Operator) {
 	case selection.Equals, selection.DoubleEquals:
 		if len(req.Values) == 0 {
 			return query.NewMatchAllQuery(), nil
 		}
+
+		if req.Key == resource.SEARCH_FIELD_ID {
```

Comment:
rather than re-constructing the document id, does it make more sense to search by `SEARCH_FIELD_NAME`?  (yes the k8s NAME ~= grafana UID is confusing!)

Reply:
name isn't being indexed. i thought we could do this rather than indexing another field making the index bigger, when the id has the value already that we can filter on